src/rgw/rgw_rest.cc

```diff
@@ -779,6 +779,29 @@ void dump_range(struct req_state *s, uint64_t ofs, uint64_t end,
 
 int RGWGetObj_ObjStore::get_params()
 {
+  // start gettorrent
+  const char *query_str = s->info.env->get("QUERY_STRING");
+  if (0 == strcmp(query_str, GET_TORRENT))
+  {
+    if ((g_conf->torrent_tracker).empty() && (g_conf->torrent_origin).empty())
+    {    
+      INFO "add torrent_tracker and torrent_origin into ceph.conf" << dendl;
+      return -1;
+    }    
+
+    torrent.set_flag(true);
+    torrent.set_info_piece_length(g_conf->sha_unit);
+    string comment(query_str);
+    torrent.set_comment(g_conf->torrent_comment);
+    torrent.set_create_by(g_conf->torrent_createby);
+    torrent.set_announce(g_conf->torrent_tracker);
+    torrent.set_encoding(g_conf->torrent_encoding);
+    torrent.set_origin(g_conf->torrent_origin);
+    string name = obj.get_orig_obj();
+    torrent.set_info_name(name);
```

Comment:
I need the object name of torrent data beacuse there is a field in torrent file format that is called name。 For example： There is a object named 1.MP4 and when you get its torrent data, a torrent file named 1.MP4.torrent will be produced.But in 1.MP4.torrent, there is a filed name to save the source file. I need get this name when put object and save it into torrent data when producing torrent data. I do not need get the name when get object. If I do not get the object name in geting object, where can I get it? thanks!